### ISSUE TITLE
WiinUSoft and WiinUPro not connecting to Official Pro Controller

### ISSUE THREAD

**Issue by OsoaGitHub** [KeyPuncher/WiinUPro#69]
The software was working perfectly until yesterday. [Here](https://streamable.com/m8i8ag) is a video of what happens. Refreshing does nothing. I've reinstalled the app and SCP driver however there is no fix.

**Comment by OsoaGitHub** [comment]
This is working on an extra computer I have, meaning it must either be a problem with the app or some sort of driver issue.

**Comment by KeyPuncher** [comment]
So that popup just closes on its own after it finds the controller; I'm thinking it's running into an error when attempting the pairing process. We'll need to surface that error to find out what's up.

**Comment by OsoaGitHub** [comment]
> 
> 
> So that popup just closes on its own after it finds the controller; I'm thinking it's running into an error when attempting the pairing process. We'll need to surface that error to find out what's up.

Thank god for the response lol, was starting to lose hope with this. How can we do this? Afaik logs aren't supported (atleast not in WiinUSoft)

**Comment by KeyPuncher** [comment]
I would have to add some code since I'm not showing every error. I should probably do more to help troubleshooting...

Anyway, I was comparing the code from WiinUSoft to WiinUPro, and because I was silly and didn't share the code, they are a little different. Are you by chance able to sync the controller using WiinUPro instead?

**Comment by OsoaGitHub** [comment]
> I would have to add some code since I'm not showing every error. I should probably do more to help troubleshooting...
> 
> Anyway, I was comparing the code from WiinUSoft to WiinUPro, and because I was silly and didn't share the code, they are a little different. Are you by chance able to sync the controller using WiinUPro instead?

Nope, they both do the exact same thing (sync menu closes instantly)

**Comment by KeyPuncher** [comment]
Okay, I'm writing some code to surface up errors.
Does that computer have more than one Bluetooth device by chance?

**Comment by OsoaGitHub** [comment]
> 
> 
> Okay, I'm writing some code to surface up errors.
> Does that computer have more than one Bluetooth device by chance?

It is paired to two other controllers but not connected.

**Comment by OsoaGitHub** [comment]
Can I get an ETA for this logging update?

**Comment by KeyPuncher** [comment]
I got most of the code written last weekend but didn't get to finish testing it. I was able to sneak some time in today. I think I found an issue, but I'm not sure how to resolve it yet, however it's not preventing my controller from connecting and I only saw it on the Wii U Pro controller once.

Hopefully the logging here will spit out a useful error code. One other thing, I noticed while connecting a Switch Pro Controller (so it may not apply) but I found that I had to click on the Windows notification that appeared to confirm and allow it to connect.

[Preview.zip](https://github.com/KeyPuncher/WiinUPro/files/6794718/Preview.zip)

**Comment by OsoaGitHub** [comment]
> 
> 
> I got most of the code written last weekend but didn't get to finish testing it. I was able to sneak some time in today. I think I found an issue, but I'm not sure how to resolve it yet, however it's not preventing my controller from connecting and I only saw it on the Wii U Pro controller once.
> 
> Hopefully the logging here will spit out a useful error code. One other thing, I noticed while connecting a Switch Pro Controller (so it may not apply) but I found that I had to click on the Windows notification that appeared to confirm and allow it to connect.
> 
> [Preview.zip](https://github.com/KeyPuncher/WiinUPro/files/6794718/Preview.zip)

Ok, I downloaded this build and tried connecting. Here is what the sync menu said:
`Searching...
Found Nintendo RVL-CNT-01-UC
Sending Pairing Code... 
Pairing Not Completed Code  (Services 1)
Setting HID Service...
Successfully Paired!
`

After this the sync menu stayed open and nothing happened, the controller didn't show up in the list or anything.

**Comment by OsoaGitHub** [comment]
Ok so I tried connecting it to my other computer (the one that it works on) today and it had the exact same log messages but it did connect

**Comment by KeyPuncher** [comment]
I had the window not close automatically so you wouldn't have to rush to get the logs.
Just looked at the code again, and I had a typo, so when it says _Pairing Not Complete Code_  without an error number following it, it was actually just a status update and didn't actually fail.

So it seems like Windows is saying everything is okay. What version of Windows are these computers?
Easy way to find the full version is to type _System Information_ in the windows search bar.

Also what does your _Device Manager_ look like?

**Comment by OsoaGitHub** [comment]
My main PC (one controller isn't working on) is version 10.0.19042 Build 19042
My spare laptop is version 10.0.19041 Build 19041

Here is what the HID-compliant game controller information says in Device Manager:
```
Device HID\{00001124-0000-1000-8000-00805f9b34fb}_VID&0002057e_PID&0330\8&377ec939&22&0000 was configured.

Driver Name: input.inf
Class Guid: {745a17a0-74d3-11d0-b6fe-00a0c90f57da}
Driver Date: 06/21/2006
Driver Version: 10.0.19041.868
Driver Provider: Microsoft
Driver Section: HID_Raw_Inst.NT
Driver Rank: 0xFF1003
Matching Device Id: HID_DEVICE_SYSTEM_GAME
Outranked Drivers: input.inf:HID_DEVICE:00FF1005
Device Updated: false
Parent Device: BTHENUM\{00001124-0000-1000-8000-00805f9b34fb}_VID&0002057e_PID&0330\7&2b207ffd&0&40F407CAD2D8_C00000000
```

This is the status on both controllers in Device Managers:
`This device is working properly.`

**Comment by KeyPuncher** [comment]
Something strange is definitely going on here. Seems very unlikely that the single revision difference would be the issue.

Try uninstalling the controller from the device manage then restarting your computer and trying again.

**Comment by OsoaGitHub** [comment]
Hey sorry for late reply, I was camping.
I've done this, still no dice. While setting up it said 'Bluetooth Periphral Device' before switching back to Nintendo RVL-CNT-01-UC.

**Comment by KeyPuncher** [comment]
Hey, sorry. I'm sure there's a solution somewhere, but I just not sure how to get there from here.

If you're feeling up to it, you can try installing the Toshiba Bluetooth stack and connect your controller through that way. https://sites.google.com/site/wiinupro/tutorials/toshiba-stack

**Comment by OsoaGitHub** [comment]
Hmm... alright. Thanks for all the help. Do you think it would be worth contacting Microsoft? If this is a problem with my computer, maybe they would know something. If I do, is there anything specific I should say?

**Comment by KeyPuncher** [comment]
They might be helpful if you can't connect any other Bluetooth devices as well.
Also, have you tried to connect the controller via the normal Windows flow without WiinUPro?

One other thing is the steps listed in this post https://wiinupro.boards.net/thread/224/solve-failed-pair

**Comment by OsoaGitHub** [comment]
I can connect other devices, so I won't bother contacting them + I have already tried the steps listen in the post.

After doing some more digging, is it possible that this error that I'm getting with SCPUser.exe is related and somehow causing the issue in WiinUPro?
When opening SCPUser.exe in the Win64 folder, I get this error:
```
The application was unable to start correctly (0xc0000142)
Click OK to close the application.
```

**Comment by OsoaGitHub** [comment]
Can you give me an answer to my question please?

**Comment by KeyPuncher** [comment]
Perhaps. I would try uninstalling the SCP Driver and then restarting the computer to see if that changes anything. I mistake some have made during installation is leaving the SCP Bluetooth Driver box checked an accidentally installing that as well which isn't needed.

If I don't reply it isn't from neglect. I'm often busy and seldom have spare time. Otherwise we wouldn't be 8 years later and still not have version 1.

**Comment by OsoaGitHub** [comment]
Yeah that also didn't work. I have a feeling it's a dll / registry issue. I tested SCPUser on the PC which WiinUSoft does work on and it opened fine. Do you know who makes the SCP Driver so I can contact them?

**Comment by KeyPuncher** [comment]
If attempting to uninstall the SCP Bluetooth driver didn't do anything then it likely has nothing to do with the SCP Driver.
If you think it's a driver thing try searching the registry, perhaps for VID&0002057e_PID&0330\8&377ec939&22&0000

An old trick to fix a slightly different issue is the pair the controller with the Wii U again.
Could be something weird happened with the Bluetooth Adapter, could try re-installing it's driver, or trying a USB dongle.

**Comment by OsoaGitHub** [comment]
> 
> 
> If attempting to uninstall the SCP Bluetooth driver didn't do anything then it likely has nothing to do with the SCP Driver.
> If you think it's a driver thing try searching the registry, perhaps for VID&0002057e_PID&0330\8&377ec939&22&0000
> 
> An old trick to fix a slightly different issue is the pair the controller with the Wii U again.
> Could be something weird happened with the Bluetooth Adapter, could try re-installing it's driver, or trying a USB dongle.

hey, sorry for the long delay. i kinda forgot about this
i've tried reconnecting it to the wii u and searching the registry just makes it not respond.
i'm assuming this is the end of the line
